src/components/common/ButtonLink.tsx

```diff
@@ -0,0 +1,9 @@
+import * as React from 'react';
+import {
+    Link as RouterLink,
+    LinkProps as RouterLinkProps
+} from 'react-router-dom';
+
+export const ButtonLink = React.forwardRef<HTMLAnchorElement, RouterLinkProps>(
+    (props, ref) => <RouterLink innerRef={ref} {...props} />
```

Comment:
QQ -- does the order have to be the same? In the generics declaration, you have `<HTMLAnchorElement, RouterLinkProps`, but then in the arguments, you have `(props, ref)`, where I would expect `(ref, props)` based on the declaration.

Reply:
The order in the generic isn't necessarily related to the order of the arguments in the function. In the generic definition, the first argument is the type of the component (and thus the ref), and the second argument is the type of the props object.